labgrid/util/helper.py

```diff
@@ -51,33 +65,67 @@ def check_output(self, command, *, print_on_silent_log=False):
         if print_on_silent_log and logger.getEffectiveLevel() > ProcessWrapper.loglevel:
             self.enable_print()
 
+        if stdin_r is not None:
+            os.close(stdin_r)
+
         # close sfd so we notice when the child is gone
         os.close(sfd)
         # get a file object from the fd
         buf = b""
+        read_fds = [mfd]
+        write_fds = []
+        except_fds = []
+        if stdin_w is not None:
+            write_fds.append(stdin_w)
+
         while True:
-            try:
-                raw = os.read(mfd, 4096)
-            except BlockingIOError as ex:
-                # wait for new data and retry
-                select.select([mfd], [], [mfd], 0.1)
-                continue
-            except OSError as e:
-                if e.errno == 5:
-                    break
-                raise
-
-            if raw:
-                buf += raw
-                *parts, buf = buf.split(b'\r')
-                res.extend(parts)
-                for part in parts:
-                    for callback in self.callbacks:
-                        callback(part, process)
+            ready_r, ready_w, ready_x = select.select(read_fds, write_fds, except_fds, 0.1)
+
+            if mfd in ready_r:
+                raw = None
+
+                try:
+                    raw = os.read(mfd, 4096)
+                except BlockingIOError as e:
```

Comment:
```suggestion
                except BlockingIOError:
```
Drop another unused variable.

Reply:
Done